src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -713,9 +716,14 @@ private void AddEasyHandle(RequestCompletionSource state)
                         throw new HttpRequestException(SR.net_http_client_execution_error,
                             GetCurlException(result, true));
                     }
+                    state.SessionHandle = _multiHandle;
+                    if (null == _multiHandle.FdSet)
+                    {
+                        _multiHandle.FdSet = new HashSet<int>();
+                        PollFunction(_multiHandle);
```

Comment:
Sorry, I didn't understand the question. When last handle is removed, we are cancelling in RemoveEasyHandle which should terminate the polling. When count goes up to 1 again, a new instance of the loop will be started.